### ISSUE TITLE
Communities: Share All Account Address

### ISSUE THREAD

**Issue by FFFra** [status-im/status-react#18012]
# Feature Issue

[Figma Design](https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=15331-288514&mode=design&t=11ZK4xcYm0njxRrt-0)

### User Story

- As an existing community member
- I want new accounts to be automatically shared with the community
- So that I don’t need to take any future action

![image](https://github.com/status-im/status-mobile/assets/5999878/8e8685ca-f8ba-4f59-aa30-bed22b33d818)

### Description

This feature aims to automate the process of sharing new accounts with the existing community. Whenever a new account is created, it should be automatically visible or shared with all existing community members. This process should be seamless and require no additional actions from current members. The feature should be integrated in a way that respects privacy and security standards.

### Acceptance Criteria

- Follow Figma
- New accounts should be automatically shared with all existing community members upon creation.
- The process should require no manual action from either new or existing members.
- The feature should be tested for both public and private community settings.

Depends on: https://github.com/status-im/status-mobile/issues/17969

### Tasks
status-go: 
  - [ ] Update Join Community API/Implement New API for the Toggle
  - [ ] Implement logic to automatically Share Future Addresses if the feature is enabled.

status-mobile:
  - [x] #18846

**Comment by ajayesivan** [comment]
It seems desktop haven't implemented this feature yet,

<img width="1512" alt="Screenshot 2023-12-14 at 18 25 36" src="https://github.com/status-im/status-mobile/assets/19279756/7e976360-b8c0-4a24-ad24-b37bcf8ef6d0">

**Comment by ajayesivan** [comment]
This functionality is not currently implemented in Desktop. Also, we need to implement the status-go side.

So for now, [as discussed with Andrea](https://discord.com/channels/@me/1176849358147293194/1186185506078347274) it would be best to descope this feature for now.

**Comment by churik** [comment]
I think for now it can be descoped

**Comment by ilmotta** [comment]
Linking with https://github.com/status-im/status-mobile/issues/18977. In case we close due to descoping that one this will be closed too.

**Comment by ilmotta** [comment]
Closing because this feature was implemented in:

- https://github.com/status-im/status-go/pull/5354
- https://github.com/status-im/status-mobile/pull/20549
